demo_usage/cuboids_to_bboxes.py

```diff
@@ -216,40 +238,64 @@ def dump_clipped_3d_cuboids_to_images(
                         copy.deepcopy(camera_config),
                     )
 
-                cv2.imwrite(save_img_fpath, img)
-                saved_img_fpaths += [save_img_fpath]
-                if max_num_images_to_render != -1 and len(saved_img_fpaths) > max_num_images_to_render:
+                if generate_video_only:
+                    video_writer.add_frame(img[:, :, ::-1])
+                else:
+                    cv2.imwrite(save_img_fpath, img)
+                    saved_img_fpaths += [save_img_fpath]
+                if (
+                    not generate_video_only
+                    and max_num_images_to_render != -1
+                    and len(saved_img_fpaths) > max_num_images_to_render
+                ):
                     flag_done = True
                     break
+            if generate_video_only:
+                video_writer.complete()
+                ffmpeg_compress_video(mp4_path, fps)
             if flag_done:
                 break
-        category_subdir = "amodal_labels"
-
-        if not Path(f"{experiment_prefix}_{category_subdir}").exists():
-            os.makedirs(f"{experiment_prefix}_{category_subdir}")
 
-        for cam_idx, camera_name in enumerate(RING_CAMERA_LIST + STEREO_CAMERA_LIST):
-            # Write the cuboid video -- could also write w/ fps=20,30,40
-            if "stereo" in camera_name:
-                fps = 5
-            else:
-                fps = 30
-            img_wildcard = f"{save_dir}/{camera_name}_%*.jpg"
-            output_fpath = f"{experiment_prefix}_{category_subdir}/{log_id}_{camera_name}_{fps}fps.mp4"
-            write_nonsequential_idx_video(img_wildcard, output_fpath, fps)
+        if not generate_video_only:
+            for cam_idx, camera_name in enumerate(RING_CAMERA_LIST + STEREO_CAMERA_LIST):
+                # Write the cuboid video from individual frames -- could also write w/ fps=20,30,40
+                fps = fps_map[camera_name]
+                img_wildcard = f"{save_dir}/{camera_name}_%*.jpg"
+                output_fpath = f"{video_output_dir}/{log_id}_{camera_name}_{fps}fps.mp4"
+                write_nonsequential_idx_video(img_wildcard, output_fpath, fps)
 
     return saved_img_fpaths
 
 
 def main(args: Any):
     """Run the example."""
     log_ids = [log_id.strip() for log_id in args.log_ids.split(",")]
-    dump_clipped_3d_cuboids_to_images(
-        log_ids,
-        args.max_num_images_to_render * 9,
-        args.dataset_dir,
-        args.experiment_prefix,
-    )
+    if args.use_multiprocessing:
+        single_process_args = [
+            (
+                [log_id],
+                args.max_num_images_to_render * 9,
+                args.dataset_dir,
+                args.experiment_prefix,
+                not args.no_motion_compensation,
+                args.omit_centerlines,
+                args.generate_video_only,
+            )
+            for log_id in log_ids
+        ]
+        with Pool(os.cpu_count()) as p:
```

Comment:
Since it take awhile to run, would be nice to be able to specify number of cpu to use as well so we can do something else while rendering.